### PR TITLE
Add code for gpio control with udev and some other things

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**
Connects-to: #22
Change-type: minor

Hi @tmigone . I wrote with @johnborgesfl in his issue. He asked for code like this and i have now this. 
I think everyone needs his own specific code so should my python script be usefull.

- [x] Add python scripts which start on disconnect and connect
- [x] Install gpiozero
- [x] Delete supervisor api label from compose
- [x] Add discovery message to airplay

suggestions: 
 - [x] https://github.com/balena-io-projects/balena-sound/pull/56#issuecomment-555672510

### CODE REVIEW

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by chrisys** [comment]
@AlexProgrammerDE this is great work - thank you for the contribution! Can you add some instructions to the readme for anyone that wants to use this?

I'd propose changing the configuration variables `BLUETOOTH_CONNECT` and `BLUETOOTH_DISCONNECT` to something like `ENABLE_SCRIPTS` to make it more obvious what they are doing, and check for `true`. I don't think it needs two separate variables, and I don't think it's clear to set them to any value - it's easier for users to follow instructions if we say exactly what it should be set to.

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
Hi @chrisys . I now added a README info and changed the two variables to `BLUETOOTH_SCRIPTS`.

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by chrisys** [comment]
@tmigone have you taken a look at this PR?

**Comment by tmigone** [comment]
Not yet @chrisys, will do tomorrow most likely.

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by chrisys** [comment]
@AlexProgrammerDE could you send me an email so I can get in touch with you? chriscw@balena.io

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by tmigone** [comment]
Hey @AlexProgrammerDE, I'm looking into this PR now, would you mind squashing your commit history into a single commit? This is to avoid spamming the master branch. 

Here are some useful links in case you are not familiar with `git rebase`:
- https://www.devroom.io/2011/07/05/git-squash-your-latests-commits-into-one/
- https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@tmigone i think i now finished with squashing

File: bluetooth-audio/bluetooth-udev [hunk 1]
```diff
@@ -16,10 +16,16 @@ add)
     echo $NAME | tr -d '"' > /var/cache/bluetooth/reconnect_device  # Save the mac address of the last connected device (for reconnect after reboot)
     mplayer -volume $CONNECTION_NOTIFY_VOLUME /usr/src/sounds/connect.mp3 -really-quiet > /dev/null 2>&1
     amixer sset PCM,0 $SYSTEM_OUTPUT_VOLUME% > /dev/null &
+    if [[ "$BLUETOOTH_SCRIPTS" == true ]]; then
```
Comment (tmigone) [inline bluetooth-audio/bluetooth-udev]: I would change this line to check if the variable is defined or not, so something like this:

```bash
if [[ -z "$BLUETOOTH_SCRIPTS" ]]; then
   python /usr/src/bluetooth-scripts/bluetooth-connect
fi
```

File: README.md [hunk 2]
```diff
@@ -62,6 +62,13 @@ By default, balenaSound bluetooth will connect using Secure Simple Pairing mode.
 
 **Note**: Legacy Mode is no longer allowed on [iOS](https://developer.apple.com/accessories/Accessory-Design-Guidelines.pdf) devices.
 
+### Bluetooth Scripts
+
+By default, balenaSound will run no scripts at connect and disconnect of a bluetooth device. If you would like to activate this, set the `BLUETOOTH_SCRIPTS` environment variable to `true`.
+There are two python scripts in the `./bluetooth-audio/bluetooh-scripts/` directory, which run on connect and disconnect. If you would like to control over GPIO a external amplifier can you use gpiozero which is preinstalled.
```
Comment (tmigone) [inline README.md]: In here I would rephrase it so that we get across this two points:
- Sample scripts are provided in `./bluetooth-audio/bluetooth-scripts/` directory
- You should/can edit them to your linking

File: README.md [hunk 3]
```diff
@@ -62,6 +62,13 @@ By default, balenaSound bluetooth will connect using Secure Simple Pairing mode.
 
 **Note**: Legacy Mode is no longer allowed on [iOS](https://developer.apple.com/accessories/Accessory-Design-Guidelines.pdf) devices.
 
+### Bluetooth Scripts
+
+By default, balenaSound will run no scripts at connect and disconnect of a bluetooth device. If you would like to activate this, set the `BLUETOOTH_SCRIPTS` environment variable to `true`.
+There are two python scripts in the `./bluetooth-audio/bluetooh-scripts/` directory, which run on connect and disconnect. If you would like to control over GPIO a external amplifier can you use gpiozero which is preinstalled. 
+
+**Note:** Don´t forget to go back to the root directory of this project before you push it in the next step.
```
Comment (tmigone) [inline README.md]: I don't think this clarification is needed :P

Comment (tmigone) [reply README.md]: I'm referring to the `Note` section here.

File: bluetooth-audio/bluetooth-udev (hunk 1 above)
Comment (tmigone) [reply bluetooth-audio/bluetooth-udev]: Same with the disconnect case.

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@tmigone i now added the from you requested things.

File: README.md [hunk 4]
```diff
@@ -62,6 +62,11 @@ By default, balenaSound bluetooth will connect using Secure Simple Pairing mode.
 
 **Note**: Legacy Mode is no longer allowed on [iOS](https://developer.apple.com/accessories/Accessory-Design-Guidelines.pdf) devices.
 
+### Bluetooth Scripts
```
Comment (tmigone) [inline README.md]: Hey @AlexProgrammerDE , how about something like this? Just trying to keep the wording consistent. Thanks!

### Bluetooth Scripts (optional)

balenaSound has configurable scripts you can run on connect and disconnect bluetooth events. If you would like to activate this, set the `BLUETOOTH_SCRIPTS` environment variable to `true`. Sample scripts can be found on the `./bluetooth-audio/bluetooth-scripts/` directory, theses can be edited as needed.

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@tmigone i now finished.

File: bluetooth-audio/Dockerfile.template [hunk 5]
```diff
@@ -10,17 +10,19 @@ RUN install_packages \
...
 
-# Setup udev rules - this lets us play the connect/disconnect sound
-# and also turn off discover/pairing when a client is connected
+# Setup udev rules - this lets us play the connect/disconnect sound,
+# turn off discover/pairing when a client is connected
+# and also run a python script
 COPY bluetooth-udev /usr/src/
 RUN chmod +x /usr/src/bluetooth-udev
 COPY udev-rules/ /etc/udev/rules.d/
+COPY udev-python/ /usr/src/
```
Comment (tmigone) [inline bluetooth-audio/Dockerfile.template]: Last change, this line is outdated, should now be:
`COPY bluetooth-scripts/ /usr/src/bluetooth-scripts/`

File: bluetooth-audio/Dockerfile.aarch64 [hunk 6]
```diff
@@ -10,17 +10,19 @@ RUN install_packages \
...
 
-# Setup udev rules - this lets us play the connect/disconnect sound
-# and also turn off discover/pairing when a client is connected
+# Setup udev rules - this lets us play the connect/disconnect sound,
+# turn off discover/pairing when a client is connected
+# and also run a python script
 COPY bluetooth-udev /usr/src/
 RUN chmod +x /usr/src/bluetooth-udev
 COPY udev-rules/ /etc/udev/rules.d/
+COPY udev-python/ /usr/src/
```
Comment (tmigone) [inline bluetooth-audio/Dockerfile.aarch64]: And here too:
`COPY bluetooth-scripts/ /usr/src/bluetooth-scripts/`

**Comment by tmigone** [comment]
Thanks for the speedy response @AlexProgrammerDE. I added one last change due to a rename that happened earlier. With that in place ill merge the PR.

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@tmigone i am finished. You can now merge.
